ui/lib/kv/addon/routes/secret/details.js

```diff
@@ -18,16 +18,19 @@ export default class KvSecretDetailsRoute extends Route {
 
   model(params) {
     const parentModel = this.modelFor('secret');
-    if (params.version) {
+    // Only fetch versioned data if selected version does not match parent (current) version
```

Comment:
I had the conditional like this originally but this means that if any changes happen to the current version they won't be captured if we click back to that version since the secret is pulled from the store, not re-queried. 

Maybe it's an edge case? But if a user elsewhere deleted/destroyed that secret version, or does so on the CLI, the UI wouldn't update unless the user exited the route or did a hard refresh

Reply:
The metadata is also pulled from the store, so we'll need some way to refresh the base `secret` route data either way. I think minimizing requests by default and then having some way to refresh is ideal. Maybe we can even set some sort of staleness on the original query so that it re-fetches if it's older than some amount of time